compose_rl/utils/mlflow_utils.py

```diff
@@ -0,0 +1,167 @@
+    """Return True if artifact_path exists (file or directory) for the run.
+
+    Artifact path needs to be a relative path to the save folder.
+    """
+    client = mlflow.MlflowClient()
+    run_id = get_mlflow_run_id()
+    assert run_id is not None, "Run ID must be set"
+
+    # Walk down the path parts level-by-level
+    parent = ""
+    if artifact_path:
+        parts = artifact_path.split("/")
+        for i, part in enumerate(parts):
+            entries = {os.path.basename(fi.path): fi for fi in client.list_artifacts(run_id, parent)}
+            if part not in entries:
+                return False
+            fi = entries[part]
+            is_last = (i == len(parts) - 1)
+            if not is_last and not fi.is_dir:
+                # trying to descend into a file
+                return False
+            parent = fi.path  # descend
+
+    # If we got here, the path exists (root or found item).
+    return True
+
+
+def get_valid_mlflow_experiment_name(config: Any) -> str:
+    """Fixes the experiment name to be an absolute path for mlflow.
+
+    MLflow requires the experiment name to be an absolute path.
+    If the experiment name is not an absolute path, we turn it
+    into an absolute path.
+    """
+    mlflow_experiment_name = config.loggers.mlflow.experiment_name
+    if mlflow_experiment_name.startswith('/'):
+        return mlflow_experiment_name
+    else:
+        from databricks.sdk import WorkspaceClient
+        return f'/Users/{WorkspaceClient().current_user.me().user_name}/{mlflow_experiment_name}'
+
+
+def get_mlflow_run_name(config: Any) -> str:
+    """Gets the mlflow run name from the config.
+
+    If the run name is not set in the config, it will return the COMPOSER_RUN_NAME environment variable
+    as this is set for interactive mode as well.
+    """
+    try:
+        return config.loggers.mlflow.tags.run
+    except:
+        return os.environ['COMPOSER_RUN_NAME']
+
+# NOTE: This doesn't work yet for a few reasons:
+# 1. Downloading nested mlflow artifacts doesn't work correctly due to the MlflowObjectStore
+# having issues. For instance, https://github.com/mosaicml/composer/blob/4ae29b1afec56ce2d54f6fa07a7f9578a0d364b0/composer/utils/object_store/mlflow_object_store.py#L465-L476
+# requires `tmp_path = os.path.join(tmp_dir, os.path.basename(artifact_path))` instead of what it currently
+# does. By doing that, the symlink can be loaded correctly.
+# 2. If save_folder is an absolute path (e.g. /tmp/checkpoints), the symlink will be created using this
+# absolute path. This is not a valid symlink in mlflow so we need to do some os.path gymnastics to
+# support absolute paths for save_folder.
+# 3. We also need to support save_folder being a dbfs path eventually.
+# Proposed Approach
+# - Create an MlflowCheckpointActor (allowing us to set WORLD_SIZE=1)
+# and create functions within that are based on MlflowObjectStore.
+# that safely handle dbfs paths and absolute paths.
```

Comment:
This PR is NOT ready for review since there's a lot of `os.path` gymnastics that we are doing for supporting saving things to mlflow artifacts. I am going to keep this PR on hold for now until we have time to think of a more resilient solution that addresses the problems here. (cc: @irenedea @bowenyang008)